### ISSUE TITLE
openxlsx::saveWorkbook(): zip error

### ISSUE THREAD

**Issue by AndreM84** [r-lib/zip#46]
Hi folks,

got trouble with the openxlsx-package after several package updates.
The issue was already posted in the related github repository. However, no helping answers so far.
As far as I know, openxlsx relies on the zip() function of r-lib and the error code seems to be related only to the call of the zip-package.
Here is the [issue in opxlsx-repository](https://github.com/ycphs/openxlsx/issues/40)

**Describe the bug**
When trying to save a workbook variable to an .xlsx file I receive the following error message:

> Fehler in zip_internal(zipfile, files, recurse, compression_level, append = FALSE, :
> zip error: Cannot add file [Content_Types].xmlto archivefile347c2455470.xlsx`` in file zip.c:394

This is an example oc code that leads to the error
```
# RScript for 'openxls saveWorkbook'-Issue:
library(openxlsx)
# Load dataset 'mtcars'
var1 <- mtcars
# Create workbook variable
wb1 <- createWorkbook("wb_mtcars")
addWorksheet(wb = wb1, sheetName = names(var1)[1])
writeData(wb=wb1, sheet = names(var1)[1], x = var1[1] ,withFilter = FALSE)
saveWorkbook(wb=wb1, file=paste(getwd(), '/wb1.xlsx',sep = ""),overwrite = TRUE)
```

System environmental variables seem to be ok and writing on the AppDat/Local directory seems to be ok as well.
Any idea, where I could continue to find a reason for this error?`

Please let me know if any additional information is required.

Just for information purposes, [here ](https://stackoverflow.com/questions/60154634/cannot-save-excel-file-in-rstudio-after-binding-data/60299438#60299438)is another (maybe) similar case.

**Comment by AndreM84** [comment]
Hi,
I am still working on the issue, but I think now that the openxlsx function _saveworkbook()_ does not work with your _zip()_ function. The error message refers to _zip.c_. Is this another function in another package?

If so, please excuse, that I opened this issue in the wrong repository.

> Fehler in zip_internal(zipfile, files, recurse, compression_level, append = FALSE, :
> zip error: Cannot add file [Content_Types].xmlto archivefile4b0c2ce76611.xlsx`` in file zip.c:394
> 
> function (..., call. = TRUE, immediate. = FALSE, noBreaks. = FALSE,
> domain = NULL)
> {
> args <- list(...)
> if (length(args) == 1L && inherits(args[[1L]], "condition")) {
> cond <- args[[1L]]
> if (nargs() > 1L)
> cat(gettext("additional arguments ignored in warning()"),
> "\n", sep = "", file = stderr())
> message <- conditionMessage(cond)
> call <- conditionCall(cond)
> withRestarts({
> .Internal(.signalCondition(cond, message, call))
> .Internal(.dfltWarn(message, call))
> }, muffleWarning = function() NULL)
> invisible(message)
> }
> else .Internal(warning(call., immediate., noBreaks., .makeMessage(...,
> domain = domain)))
> }
> <bytecode: 0x000001602a238488>
> <environment: namespace:base>

If this is the correct repository, could the following help to solve the issue?
I found an interesting issue regarding encoding within the read_ and write_ functions in the tidyverse package on their github repository ([link](https://github.com/tidyverse/readr/issues/697)) and I was wondering if such a missing encoding information in the zip function of r-lib might cause the described errors on my machine.

Could you just let me know if this is the rigth repository?

**Comment by annececile22** [comment]
Hi André, 
I am facing the same issue. Have you found out a solution? Thanks!
Anne-Cécile

**Comment by gaborcsardi** [comment]
I can't reproduce this, unfortunately. If you still see it, can you please send the output of `sessioninfo::session_info()`? Thanks!

**Comment by FrancoisR95** [comment]
Hello, I have exactly the same bug and **here is how to reproduce it**.

On my system (Windows 10), it occurs for instance when using openxlsx version 4.1.5 (which uses zipr) but not with openxlsx version 4.1.2 (which uses zip), so my workaround with openxlsx is to downgrade to 4.1.2.

But the bug is really in zip, not in openxlsx, as I can reproduce it directly with zip or zipr (version 2.0.4) with the following code:

<pre><code>library(zip)

# --- create 1 temporary directory and 2 test sub-directories ----
dir.create(tmpdir <- tempfile())
dir.create("C:/Test/")
dir.create(dir_no_accent <- "C:/Test/dir")
dir.create(dir_with_accent <- "C:/Test/diré")

# --- create 2 files in each of these 3 directories ----
cat("first file", file = file.path(tmpdir, "file1"))
cat("second file", file = file.path(tmpdir, "file2"))
cat("first file", file = file.path(dir_no_accent, "file1"))
cat("second file", file = file.path(dir_no_accent, "file2"))
cat("first file", file = file.path(dir_with_accent, "file1"))
cat("second file", file = file.path(dir_with_accent, "file2"))

# --- create path for temporary zip file ----
tmpzipfile <- tempfile(fileext = ".zip")
# --- create path for zip file without accent ----
zipfile_no_accent <- "C:/Test/zipfile.zip"
# --- create path for zip file with accent ----
zipfile_with_accent <- "C:/Test/zipfilé.zip"

# --- (1) Using zip & zipr with tmpdir (fails for both zip and zipr) ----
zipr(tmpzipfile, tmpdir)
zip(tmpzipfile, tmpdir)
# --- (2) Using zip & zipr with zipfile_with_accent & dir_with_accent (fails for both zip and zipr) ----
zipr(zipfile_with_accent, dir_with_accent)
zip(zipfile_with_accent, dir_with_accent)
# --- (3) Using zip & zipr with dir_with_accent (fails for both zip and zipr) ----
zipr(tmpzipfile, dir_with_accent)
zip(tmpzipfile, dir_with_accent)
# --- (4) Using zip & zipr with dir_no_accent (success for zipr / fails for zip) ----
zipr(tmpzipfile, dir_no_accent)
zip(tmpzipfile, dir_no_accent)
# --- (5) Using zip & zipr with zipfile_no_accent/dir_no_accent (success for zipr / fails for zip) ----
zipr(zipfile_no_accent, dir_no_accent)
zip(zipfile_no_accent, dir_no_accent)</code></pre>

Here are the error messages I get:
(1) Fails on my system because I have a "ç" in my Windows username which is therefore in the path of all tmp files and directories (certainly the same for @AndreM84 with "André" and @annececile22  with "Cécile") but users without plain ASCII usernames should not have issues there: 
<pre><code>> # --- (1) Using zip & zipr with tmpdir (fails for both zip and zipr) ----
> zipr(tmpzipfile, tmpdir)
Error in zip_internal(zipfile, files, recurse, compression_level, append = FALSE,  : 
  zip error: `Cannot add file `file317c551c6316/file1` to archive `C:\Users\FRANOI~2\AppData\Local\Temp\Rtmp6dUEqg\file317c776774c.zip`` in file `zip.c:394`
> zip(tmpzipfile, tmpdir)
Error in zip_internal(zipfile, files, recurse, compression_level, append = FALSE,  : 
  zip error: `Cannot add directory `C:\Users\FRANOI~2\AppData\Local\Temp\Rtmp6dUEqg\file317c551c6316/` to archive `C:\Users\FRANOI~2\AppData\Local\Temp\Rtmp6dUEqg\file317c776774c.zip`` in file `zip.c:387`</code></pre>
(2) Fails and will certainly fail for any user (<b>this is the easy way to reproduce the problem for anyone, at least on Windows with RStudio</b>) 
<pre><code>> # --- (2) Using zip & zipr with zipfile_with_accent & dir_with_accent (fails for both zip and zipr) ----
> zipr(zipfile_with_accent, dir_with_accent)
Error in zip_internal(zipfile, files, recurse, compression_level, append = FALSE,  : 
  zip error: `Cannot add file `dirÃ©/file1` to archive `C:/Test/zipfilé.zip`` in file `zip.c:394`
> zip(zipfile_with_accent, dir_with_accent)
Error in zip_internal(zipfile, files, recurse, compression_level, append = FALSE,  : 
  zip error: `Cannot add directory `C:/Test/diré/` to archive `C:/Test/zipfilé.zip`` in file `zip.c:387`</code></pre>
(3) Fails and will maybe fail also for any user (<b>this might also be a way to reproduce the problem for anyone</b>) 
<pre><code>> # --- (3) Using zip & zipr with dir_with_accent (fails for both zip and zipr) ----
> zipr(tmpzipfile, dir_with_accent)
Error in zip_internal(zipfile, files, recurse, compression_level, append = FALSE,  : 
  zip error: `Cannot add file `dirÃ©/file1` to archive `C:\Users\FRANOI~2\AppData\Local\Temp\Rtmp6dUEqg\file317c776774c.zip`` in file `zip.c:394`
> zip(tmpzipfile, dir_with_accent)
Error in zip_internal(zipfile, files, recurse, compression_level, append = FALSE,  : 
  zip error: `Cannot add directory `C:/Test/diré/` to archive `C:\Users\FRANOI~2\AppData\Local\Temp\Rtmp6dUEqg\file317c776774c.zip`` in file `zip.c:387`</code></pre>
(4) Fails on my system with zip but success with zipr 
<pre><code>> # --- (4) Using zip & zipr with dir_no_accent (success for zipr / fails for zip) ----
> zipr(tmpzipfile, dir_no_accent)
> zip(tmpzipfile, dir_no_accent)
Error in zip_internal(zipfile, files, recurse, compression_level, append = FALSE,  : 
  zip error: `Cannot add directory `C:/Test/dir/` to archive `C:\Users\FRANOI~2\AppData\Local\Temp\Rtmp6dUEqg\file317c776774c.zip`` in file `zip.c:387`</code></pre>
(5) Fails on my system with zip but success with zipr 
<pre><code>> # --- (5) Using zip & zipr with zipfile_no_accent/dir_no_accent (success for zipr / fails for zip) ----
> zipr(zipfile_no_accent, dir_no_accent)
> zip(zipfile_no_accent, dir_no_accent)
Error in zip_internal(zipfile, files, recurse, compression_level, append = FALSE,  : 
  zip error: `Cannot add directory `C:/Test/dir/` to archive `C:/Test/zipfile.zip`` in file `zip.c:387`</code></pre>
@gaborcsardi : are you able to reproduce the bug with the code (2) above?

To recap, on my system:
- zipr is hit by this bug whenever there is a non-ASCII character in the path of the source file(s) to compress, but it can handle accents in the path of the destination zip file.
- I was not able to make zip work on my system: it's affected by the bug even when the source file(s) and the destination zip file do not have accent in their path (maybe it still uses a tmp file somehow, which is not plain ASCII on my system).
- However openxlsx version 4.1.2 works on my system, although it is using zip. I don't know how it manages to bypass the bug.

François

**Comment by gaborcsardi** [comment]
OK, so if this is only about non-ascii characters in paths, then I'll close this issue and try to fix #53.

**Comment by FrancoisR95** [comment]
A workaround which worked for me was to change my tmp directory, as I just explained in https://github.com/r-lib/zip/issues/53.

Initially my tmp directory contained a non-ASCII character ("ç" in "François"):
````R
> tempdir()
[1] "C:\\Users\\FRANOI~2\\AppData\\Local\\Temp\\Rtmp04a928"
````
For changing the tmp directory, I created in the R home directory a Renviron.site file containing the following line: <code>TMPDIR=C:/R-home/rtmp</code>
Then, after restarting Rstudio, the tmp directory was changed:
````R
> tempdir()
[1] "C:/R-home/rtmp\\Rtmpa0TIPC"
```` 

And now, I don't have any error anymore with saveWorkbook in openxlsx 4.1.5. I had also upgraded R from 4.0.0 to 4.0.2 yesterday, which may (or not) have had an impact.
